### PR TITLE
Add isTrialOnly for subscriptions

### PR DESCRIPTION
**Pull request by lexerom**
Add isTrialOnly param for subscriptions

### CODE REVIEW

**Review by removed-account-123 — DISMISSED** [review]

File: openapi/components/schemas/Subscription/OrderTypes/subscription-order.yaml [hunk 1]
```diff
@@ -37,6 +37,10 @@ allOf:
             description: The time the trial should end
             type: string
             format: date-time
+      isTrialOnly:
+        description: isTrialOnly determines if subscription will be ended after trial
```
Comment (slavcodev) [inline openapi/components/schemas/Subscription/OrderTypes/subscription-order.yaml:41]:
```suggestion
        description: Whether a subscription should be ended after trial.
```

Comment (lexerom) [reply openapi/components/schemas/Subscription/OrderTypes/subscription-order.yaml:41]: Agree.

**Review by VladimirBerdnik — DISMISSED** [review]

**Review by slavcodev — DISMISSED** [review]

**Review by creocoder — DISMISSED** [review]

File: openapi/components/schemas/Subscription/OrderTypes/subscription-order.yaml [hunk 2]
```diff
@@ -37,6 +37,10 @@ allOf:
             description: The time the trial should end
             type: string
             format: date-time
+      isTrialOnly:
+        description: Whether a subscription churns after a trial period.
```
Comment (2e3s) [inline openapi/components/schemas/Subscription/OrderTypes/subscription-order.yaml:41]:
```suggestion
        description: Whether a subscription ends after a trial period.
```

Comment (2e3s) [reply openapi/components/schemas/Subscription/OrderTypes/subscription-order.yaml:41]:
```suggestion
        description: Whether a subscription ends after a trial period. Recurring settings are ignored if it's `true`.
```
No precision is excessive :slightly_smiling_face:

**Comment by lexerom** [comment]
I am thinking about if we add new enum param like 'recurringType' and values 'trial-only', 'recurring-only' and 'default' (or absent or null) and get rid of trial and isTrialOnly params and add trialEndTime instead. So, it will be like this:
```
recurringType: 'trial-only',
trialEndTime: 'some-valid-date'
```
instead of
```
trial: {
  enabled: true,
  endTime: 'valid-date'
},
isTrialOnly: true
```
AND

`recurringType: 'recurring-only',`

instead of
```
trial: {
  enabled: false,
  endTime: null
}
```

**Comment by creocoder** [comment]
@lexerom Looks overcomplicated. I think it better to add `isRecurringOnly` and use `trial.enabled` as BC property to fix consistency.

**Review by creocoder — APPROVED** [review]

**Review by VladimirBerdnik — APPROVED** [review]

**Pull request merged**